cluster-autoscaler/cloudprovider/aws/README.md

```diff
@@ -100,52 +100,51 @@ Auto-Discovery Setup is the preferred method to configure Cluster Autoscaler.
 To enable this, provide the `--node-group-auto-discovery` flag as an argument
 whose value is a list of tag keys that should be looked for. For example,
 `--node-group-auto-discovery=asg:tag=k8s.io/cluster-autoscaler/enabled,k8s.io/cluster-autoscaler/<cluster-name>`
-will find the ASGs where those tag keys _exist_. It does not matter what value
+makes Cluster Autoscaler look for ASGs where _all_ of these tag keys exist. It does not matter what value
 the tags have.
 
+In other words, if you have ASGs with differing (non-common) tags, you will not want to pass these tags to `--node-group-auto-discovery`, because Cluster Autoscaler [looks for ASGS with all these tag-keys](https://github.com/kubernetes/autoscaler/blob/c94ade5266a6c4515aa4a0c7365af0ffb1cf8989/cluster-autoscaler/cloudprovider/aws/aws_manager.go#L471-L472) and will fail to match _some_ or _all_ of your ASGs. Only pass tags that are common across all ASGs in your cluster.
```

Comment:
I'm not sure linking to a comment in code which is saying the same thing as this doc is adding anything here. I'd either link to the actual code which does this, or remove the link.